### PR TITLE
[WIP] Inferred sorting rule

### PR DESCRIPTION
**Pull request by marcelofabri**
Continued from  #1050

// cc @JamieEdge

### CODE REVIEW

**Comment by marcelofabri** [comment]
This should have triggered the `Rules.md` warning in Danger 🤔

File: Dangerfile [hunk 1]
```diff
@@ -3,15 +3,17 @@ require 'open3'
...
-has_danger_changes = !git.modified_files.grep(/Dangerfile|script\/oss-check|Gemfile/).empty?
-has_rules_changes = !git.modified_files.grep('/Source/SwiftLintFramework/Rules/').empty?
-has_rules_docs_changes = !git.modified_files.grep('Rules.md').empty?
+has_app_changes = !modified_files.grep(/Source/).empty?
+has_test_changes = !modified_files.grep(/Tests/).empty?
+has_danger_changes = !modified_files.grep(/Dangerfile|script\/oss-check|Gemfile/).empty?
+has_build_changes = !modified_files.grep(/Makefile|SwiftLint\.xcodeproj|SwiftLint\.xcworkspace|Package\.swift|Cartfile/).empty?
+has_danger_changes = !modified_files.grep(/Dangerfile|script\/oss-check|Gemfile/).empty?
+has_rules_changes = !modified_files.grep(/Source\/SwiftLintFramework\/Rules/).empty?
+has_rules_docs_changes = !modified_files.grep(/Rules.md/).empty?
```
Comment (jpsim) [inline Dangerfile]: you need to escape this period: `/Rules\.md/`

**Comment by marcelofabri** [comment]
@jpsim What do you think about this? I'm afraid that in practice this rule is not as helpful as I've thought it would be.

**Comment by masters3d** [comment]
I think this rule is not too helpful when applied to arrays.  
It would be somewhat helpful when used on dictionary literals sorted by the key. Another place would be Enum cases.

**Pull request closed without merging**